ambry-store/src/test/java/com.github.ambry.store/StoreFindTokenTest.java

```diff
@@ -209,11 +225,15 @@ private void compareTokens(StoreFindToken reference, StoreFindToken toCheck) {
    */
   private void doSerDeTest(StoreFindToken token) throws IOException {
     short i = numSegments > 1 ? (short) 1 : (short) 0;
-    for (; i <= 1; i++) {
+    for (; i <= 2; i++) {
       DataInputStream stream = getSerializedStream(token, i);
       StoreFindToken deSerToken = StoreFindToken.fromBytes(stream, STORE_KEY_FACTORY);
-      compareTokens(token, deSerToken);
+      compareTokens(token, deSerToken, i);
       assertEquals("SessionId does not match", token.getSessionId(), deSerToken.getSessionId());
+      if (i == 2) {
```

Comment:
Same here - if the equality test isn't matching, I want to understand why. For the "new" fields (`inclusive` etc), there will be a default value. So I'm unable to understand why they wouldn't match.